portal/server/app/route.ts

```diff
@@ -59,6 +60,17 @@ export async function GET(req: Request) {
     }
 
     const parsedUrl = getSubdomainAndPath(url, Number(portalDomainNameLength));
+    if (parsedUrl) {
+        const subdomainIsBlacklisted = await has(parsedUrl.subdomain)
+        if (subdomainIsBlacklisted) {
+            logger.info({
+                message: 'Attempt to access blacklisted subdomain',
+                blacklistedSubdomain: parsedUrl.subdomain
+            })
+            return new Response(`Subdomain ${parsedUrl.subdomain} is blacklisted.`, { status: 403 });
```

Comment:
I think 404 (not found) is a bit misleading, since the resource _can_ be found, but we decide not to serve it. 

With 403 (forbidden), we acknowledge that we found the resource, but we blocked it.

Reply:
Oh, I see. If that's the case, then sure, 404 sounds perfect for this.

Reply:
Done!